src/main/java/com/shima/patientchart/controller/PatientChartController.java

```diff
@@ -1,45 +1,127 @@
+        return new ResponseEntity<>(body, HttpStatus.NOT_FOUND);
+    }//404エラーを返す
 
-    @PostMapping("/patientcharts")
-    public ResponseEntity<CreateResponse> createName(@RequestBody CreateRequest createRequest, UriComponentsBuilder uriComponentsBuilder){
-        URI uri = uriComponentsBuilder.path("/patientcharts/{id}").buildAndExpand(1).toUri();
+
+    //POST 指定したIDが存在しない場合
+    @ExceptionHandler(value = MethodArgumentNotValidException.class)
+    public ResponseEntity<Map<String, String>> handleMethodArgumentNotValidException(
+            MethodArgumentNotValidException e, HttpServletRequest request) {
+        Map<String, String> body = Map.of(
+                "timestamp", ZonedDateTime.now().toString(),
+                "status", String.valueOf(HttpStatus.BAD_REQUEST.value()),
+                "error", HttpStatus.BAD_REQUEST.getReasonPhrase(),
+                "message", e.getBindingResult().getAllErrors().stream().map(err -> {
+                    if (err instanceof FieldError) {
+                        return String.format("%s: %s", ((FieldError) err).getField(), err.getDefaultMessage());
+                    }
+                    return err.toString();
+                }).collect(Collectors.joining(", ")),
+                "path", request.getRequestURI());
+        return new ResponseEntity<>(body, HttpStatus.BAD_REQUEST);
+    }//400エラーを返す
+
+
+    // Addressが既に登録されている場合
+    @ExceptionHandler(value = AddressAlreadyExistsException.class)
+    public ResponseEntity<Map<String, String>> handleAddressAlreadyExistsException(
+            AddressAlreadyExistsException e, HttpServletRequest request) {
+        Map<String, String> body = Map.of(
+                "timestamp", ZonedDateTime.now().toString(),
+                "status", String.valueOf(HttpStatus.BAD_REQUEST.value()),
+                "error", HttpStatus.BAD_REQUEST.getReasonPhrase(),
+                "message", e.getMessage(),
+                "path", request.getRequestURI());
+        return new ResponseEntity<>(body, HttpStatus.BAD_REQUEST);
+    }//400 エラー
+
+
+    // Nameが既に登録されている場合
+    @ExceptionHandler(value = NameAlreadyExistsException.class)
+    public ResponseEntity<Map<String, String>> handleNameAlreadyExistsException(
+            NameAlreadyExistsException e, HttpServletRequest request) {
+        Map<String, String> body = Map.of(
+                "timestamp", ZonedDateTime.now().toString(),
+                "status", String.valueOf(HttpStatus.BAD_REQUEST.value()),
+                "error", HttpStatus.BAD_REQUEST.getReasonPhrase(),
+                "message", e.getMessage(),
+                "path", request.getRequestURI());
+        return new ResponseEntity<>(body, HttpStatus.BAD_REQUEST);
+    }//400 エラー
+
+
+    //GETの実装
+    //全件取得の実装
+    @GetMapping("/patient-charts")
+    public List<PatientChart> getAllPatientChart() {
+        return patientchartService.findAllPatientChart();// Serviceの返り値を受け取る変数の型
```

Comment:
このコメントもう不要ですよね？削除しておきましょう！
```suggestion
        return patientchartService.findAllPatientChart();
```

Reply:
削除しました。